invenio_accounts/models.py

```diff
@@ -463,3 +468,153 @@ def delete_by_user(cls, method, user):
         """Unlink a user from an external id."""
         with db.session.begin_nested():
             cls.query.filter_by(id_user=user.id, method=method).delete()
+
+
+class DomainOrg(db.Model):
+    """Domain organisation."""
+
+    __tablename__ = "accounts_domain_org"
+
+    id = db.Column(db.Integer(), primary_key=True, autoincrement=True)
+
+    pid = db.Column(db.String(255), unique=True, nullable=True)
+    """Persistent identifier for organisation."""
+
+    name = db.Column(db.String(255), nullable=False)
+    """Name of organisation."""
+
+    json = db.Column(
+        json_field,
+        default=lambda: dict(),
+        nullable=False,
+    )
+    """Store additional metadata about the organisation."""
+
+    parent_id = db.Column(
+        db.Integer(), db.ForeignKey("accounts_domain_org.id"), nullable=True
+    )
+    """Link to parent organisation."""
+
+    parent = db.relationship("DomainOrg", remote_side=[id])
```

Comment:
Allows for creating hierarchy of organisations - useful again for statistics (on Zenodo the ROR hierarchy)